### PR TITLE
docs(plugins): add create-a-plugin guide

### LINKED ISSUE #2586: docs/api/plugins.md is outdated
I have noticed we have outdated informations in plugins.md

```md
You can extend Stryker with 6 plugin kinds:

export enum PluginKind {
  ConfigEditor = 'ConfigEditor',
  TestRunner = 'TestRunner',
  TestFramework = 'TestFramework',
  Transpiler = 'Transpiler',
  Mutator = 'Mutator',
  Reporter = 'Reporter',
}
```

it should be updated (and probably also some other parts)

### PR DESCRIPTION
**Pull request by simondel**
Fixes #2586

### CODE REVIEW

**Review by nicojs — APPROVED** [review]
Nice! Do you also want these docs to be added to the website? If so, we should list it in the `sidebar.json` file.

**Comment by simondel** [comment]
@nicojs Perhaps it can be merged with https://stryker-mutator.io/docs/stryker/plugins
What do you think?

**Comment by nicojs** [comment]
Awesome idea!

**Comment by nicojs** [comment]
In hindsight, maybe it's better to add this to the guides section. So add a guide for "creating plugins". Otherwise people browsing around might get overwhelmed with details 🤷‍♂️

**Comment by simondel** [comment]
Good idea!

File: docs/api/plugins.md [hunk 1]
```diff
@@ -1,17 +1,14 @@
 ---
 title: Plugins
-custom_edit_url: https://github.com/stryker-mutator/stryker4s/edit/master/docs/api/plugins.md
+custom_edit_url: https://github.com/stryker-mutator/stryker/edit/master/docs/api/plugins.md
 ---
 
 You can extend Stryker with 6 plugin kinds:
```
Comment (krisanselmo) [inline docs/api/plugins.md:6]:
```suggestion
You can extend Stryker with 3 plugin kinds:
```

**Comment by nicojs** [comment]
I've just moved it.

**Pull request merged**